cpp-client/README.md

```diff
@@ -121,6 +126,20 @@ C++ compiler and tool suite (cmake etc).
     ./tests
     ```
 
+Notes
+  (1) The standard assumptions for `Debug` and `Release` apply here.
+      With a `Debug` build you get debug information which is useful during
+      development and testing of your own code that depends on the client
+      and indirectly on these libraries.  A `Release` build gives you
+      optimized libraries that are faster and smaller but with no
+      debugging information.  Note that while in general it is expected
+      to be able to freely mix some `Debug` and `Release` code,
+      some of the dependent libraries are incompatible; in particular,
+      protobuf generates different code and code compiled for a `Release`
+      target using protobuf header files will not link against a `Debug`
+      version of protobuf.  To keep your life simpler, run a consistent set
+      for your code and all dependencies.
```

Comment:
super-nit. I'd suggest a more editorially-neutral "To keep things simple, we suggest that you run
a consistent setting for your code and all dependencies."

[i.e. "setting" instead of "set"; and "things" not "your life"]